source/adapters/level_zero/adapter.cpp

```diff
@@ -171,8 +215,36 @@ ur_adapter_handle_t_::ur_adapter_handle_t_()
 
       return;
     }
+    // Dynamically load the new L0 SysMan separate init and new EXP apis
+    // separately. This must be done to avoid attempting to use symbols that do
+    // not exist in older loader runtimes.
+#ifdef _WIN32
+    HMODULE processHandle = GetModuleHandle(NULL);
+#else
+    HMODULE processHandle = nullptr;
```

Comment:
nit: this will work, but should technically be `RTLD_DEFAULT` (which is NULL anyway, but... :P)